lib/player_ai_minimax.rb

```diff
@@ -0,0 +1,116 @@
+class PlayerAiMinimax
+  attr_reader :mark
+
+  CORNERS = [0,2,6,8]
+  OPPOSITE_CORNERS = [8,6,2,0]
+  def initialize(player_mark, board_rules)
+    @mark = player_mark
+    @rules = board_rules
+  end
+
+  def play_move(game_state)
+    @current_player = game_state.player
+    @non_current_player = game_state.opponent
+    board = game_state.game_board
+    if first_move?(board)
+      move = first_turn(board)
+    elsif second_move?(board)
+      move = second_turn(board)
+    end
+    move ||= best_move(board, game_state.player, game_state.opponent)
```

Comment:
No need to set a local `move` variable here—this can just be:
```ruby
if first_move?(board)
  first_turn(board)
elsif second_move?(board)
  second_turn(board)
else
  best_move(board, game_state.player, game_state.opponent)
end
```